### PR TITLE
docs(system-e2e): Adding a dash of docs about system-e2e

### PR DESCRIPTION
**Pull request by pshomov**
Adding a bit of docs for running System E2E tests. Follow up to #10341.

Attach a link to issue if relevant

## What

Specify what you're trying to achieve

## Why

Specify why you need to achieve this

## Screenshots / Gifs

Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -111,6 +112,57 @@ To execute the end-to-end tests affected by a change:
 yarn affected:e2e
 ```
 
+### Running System E2E tests
+
+To execute end-to-end tests via [Playwright](https://playwright.dev), run one or more files. The way to specify these, is similar as described in the docs [here](https://playwright.dev/docs/running-tests).
```
Comment (eirikurn) [inline README.md:117]: There's already docs for system e2e testing with Playwright here: https://docs.devland.is/repository/system-e2e

Let's merge this content into that and point to it from here.

File: README.md [hunk 2]
```diff
@@ -111,6 +112,57 @@ To execute the end-to-end tests affected by a change:
...
+    'application-system-api': '(source /Users/petar/src/andes/customers/si/island.is/.env.application-system-api && yarn start application-system-api)',
+    'icelandic-names-registry-backend': '(source /Users/petar/src/andes/customers/si/island.is/.env.icelandic-names-registry-backend && yarn start icelandic-names-registry-backend)',
+    'air-discount-scheme-backend': '(source /Users/petar/src/andes/customers/si/island.is/.env.air-discount-scheme-backend && yarn start air-discount-scheme-backend)',
+    'service-portal-api': '(source /Users/petar/src/andes/customers/si/island.is/.env.services-user-profile && yarn start services-user-profile)',
+    'services-sessions': '(source /Users/petar/src/andes/customers/si/island.is/.env.services-sessions && yarn start services-sessions)',
+    api: '(source /Users/petar/src/andes/customers/si/island.is/.env.api && yarn start api)'
+  },
+  mocks: 'docker run -it --rm -p 2525:2525 -p 9388:9388 -p 9372:9372 -v /Users/petar/src/andes/customers/si/island.is/infra/mountebank-imposter-config.json:/app/default.json docker.io/bbyars/mountebank:2.8.1 start --configfile=/app/default.json'
+}
```
Comment (eirikurn) [inline README.md:158]: Is the DX to run a command which prints JSON and then you have to manually copy commands from the JSON output?

**Review by eirikurn — CHANGES_REQUESTED** [review]
LGTM. Would like to move this to the testing docs. 👍

File: README.md [hunk 3]
```diff
@@ -111,6 +112,57 @@ To execute the end-to-end tests affected by a change:
...
+### Running System E2E tests with mocks
+
+Running System E2E tests which rely on mocking external dependencies is the same, except the local development environment must support wire-mocking(mountebank). To prepare such a local environment it is best to use the command that discovers the dependencies of the component(s) you will be testing and prepares configuration for running them. To get a list of the services you might need to run as well as the mountebank configuration:
+
+```bash
+yarn infra render-local-env  --service=<service-name-1> ... --service=<service-name-N> 
+```
+
+You will get output similar to this one:
+```bash
```
Comment (AndesKrrrrrrrrrrr) [inline README.md:144]:
`````suggestion
```bash
yarn infra render-local-env  --service=<service-name-1> ... --service=<service-name-N> 
```

For example, if you are running `application-system-form` you would run
```bash
yarn infra render-local-env  --service=application-system-form
```
And you would get something similar to this:
```bash
`````

File: README.md [hunk 4]
```diff
@@ -111,6 +112,57 @@ To execute the end-to-end tests affected by a change:
...
+    'air-discount-scheme-backend': '(source /Users/petar/src/andes/customers/si/island.is/.env.air-discount-scheme-backend && yarn start air-discount-scheme-backend)',
+    'service-portal-api': '(source /Users/petar/src/andes/customers/si/island.is/.env.services-user-profile && yarn start services-user-profile)',
+    'services-sessions': '(source /Users/petar/src/andes/customers/si/island.is/.env.services-sessions && yarn start services-sessions)',
+    api: '(source /Users/petar/src/andes/customers/si/island.is/.env.api && yarn start api)'
+  },
+  mocks: 'docker run -it --rm -p 2525:2525 -p 9388:9388 -p 9372:9372 -v /Users/petar/src/andes/customers/si/island.is/infra/mountebank-imposter-config.json:/app/default.json docker.io/bbyars/mountebank:2.8.1 start --configfile=/app/default.json'
+}
+```
+Each service will get the secrets it is configured to use, still you will need to run local DB setup for the services you are running - dev-init, etc. Additionally, any local dev proxies will also need to be run manually.
```
Comment (AndesKrrrrrrrrrrr) [inline README.md:160]:
```suggestion
Each service will get the secrets it is configured to use, still you will need to run local DB setup for the services you are running - dev-init, etc. Additionally, any local dev proxies will also need to be run manually (e.g. `run-xroad-proxy.sh`).
```

File: README.md [hunk 5]
```diff
@@ -111,6 +112,57 @@ To execute the end-to-end tests affected by a change:
...
+
+It is not necessary to run all the dependencies/services, only the one that will be actually used.
```
Comment (AndesKrrrrrrrrrrr) [inline README.md:162]:
```suggestion
It is not necessary to run all the dependencies/services, only the ones that will be actually used.
```

File: README.md (hunk 1 above)
Comment (AndesKrrrrrrrrrrr) [inline README.md:117]:
```suggestion
To execute end-to-end tests via [Playwright](https://playwright.dev), run one or more files. The way to specify these, is similar as described in the docs [here](https://playwright.dev/docs/running-tests). `yarn system-e2e` is a convenience script which runs `playwright test` with the `system-e2e` Playwright config.
```

**Review by AndesKrrrrrrrrrrr — COMMENTED** [review]
:clap: Great docs :clap: 

Some minor clarifications/wording.

File: README.md (hunk 2 above)
Comment (AndesKrrrrrrrrrrr) [reply README.md:158]: :sweat_smile: for now, yes

**Comment by saevarma** [comment]
@AndesKrrrrrrrrrrr  should we update this PR and finish it?

It should update the root `README` to deprecate the old end-to-end testing and add  the new system-e2d chapter, but it should only mention it high level and then reference the appropriate docs for more details, either the system-e2e/readme or the docs.devland testing.

You should merge the docs to either of those places. Please ping me if you want to update docs.devland and I will add you to Gitbook.

**Comment by andes-it** [comment]
This PR's had a 15-day siesta. Lost its spark, did it? It's making an exit in 5 days unless the stale label disappears.

**Comment by andes-it** [comment]
This PR closed its doors. Sat around for a day after the stale tag, then reckoned it's not worth the fuss.

**Pull request closed without merging**